homework/lesson_number_2/lesson_number_2/Program.cs

```diff
@@ -0,0 +1,214 @@
+
+                newArray[i] = num;
+            }
+            return newArray;
+        }
+
+        private static int FindNewLowestNumber(int[] numbers, int[] newArray)
+        {
+            //So first comparison is always set to num
+            int num = 999;
+
+            for (int e = 0; e < numbers.Length; e++)
+            {
+                if (numbers[e] < num)
+                {
+                    int lowestNumber = numbers[e];
+                    bool numberAlreadyInNewArray = CheckIfNumberAlreadyInNewArray(lowestNumber, newArray, e);
+
+                    if (!numberAlreadyInNewArray) num = lowestNumber;
+                }
+            }
+
+            return num;
+        }
+
+        private static bool CheckIfNumberAlreadyInNewArray(int lowestNumber, int[] newArray, int e)
+        {
+            bool numberAlreadyInNewArray = false;
+
+            foreach (var newArrayNum in newArray)
+            {
+                if (lowestNumber == newArrayNum)
+                {
+                    numberAlreadyInNewArray = true;
+                }
+            }
+
+            return numberAlreadyInNewArray;
+        }
+
+        static int[] AddElementEndOfArray(int[] numbers, int newNum)
+        {
+            int[] newArray = new int[numbers.Length + 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i + 1 == newArray.Length)
+                {
+                    newArray[i] = newNum;
+                    continue;
+                }
+                newArray[i] = numbers[i];
+            }
+            
+            return newArray;
+        }
+
+        static int[] AddElementStartOfArray(int[] numbers, int newNum)
+        {
+            int[] newArray = new int[numbers.Length + 1];
+ 
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i == 0)
+                {
+                    newArray[i] = newNum;
+                    continue;
+                }
+                newArray[i] = numbers[i - 1];
+            } 
+            
+            return newArray;
+        }
+
+        static int[] AddElementAnyPositionInArray(int[] numbers, int newNum, int position)
+        {
+            int[] newArray = new int[numbers.Length + 1];
+            bool newNumberWasAdded = false;
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i + 1 == position)
+                {
+                    newArray[i] = newNum;
+                    newNumberWasAdded = true;
+                    continue;
+                    
+                }
+
+                if (newNumberWasAdded)
+                {
+                    newArray[i] = numbers[i - 1];
+                    continue;
+                }
+                
+                newArray[i] = numbers[i];
+            }
+
+            return newArray;
+        }
+
+        static int[] RemoveElementStartOfArray(int[] numbers)
```

Comment:
These two methods (`RemoveElementStartOfArray` and `RemoveElementEndOfArray`) are much easier to read than their counterparts (`AddElementStartOfArray` and `AddElementEndOfArray`).  I think there is a way you could write the add methods that is similar to the way you wrote these.